### ISSUE TITLE
Custom Action Factory for Dynamically Created Class like UtterAction

### ISSUE THREAD

**Issue by naoko** [RasaHQ/rasa_nlu#2313]
**Rasa Core version**: master

**Python version**:  3.6

**Operating system** (windows, osx, ...): osx, linux

**Issue**:
I'm dynamically building all files (domain.yml, nlu.md, and stories.md) off of the data stored in a database.
I'm in need of action that gets build dynamically just like UtterAction.
Basically, all it does is the same thing (calling 3rd party webhook with different params).

To do so in the least invasive way, I found that allowing a user to specify custom `action_factory`.
Currently, `action_factory` can be read from domain.yml but choices are only local or remote.

I like to propose changes like this

**Content of domain file** :
```yaml
action_factory: path_to_your_action_factory_function
```
`rasa_core.actions.factories.action_factory_by_name` to be

```
import importlib

def action_factory_by_name(name):
    if name == "local" or name is None:  # this is the default factory
        return local_action_factory
    elif name == "remote":
        return remote_action_factory
    try:
        mod_name, func_name = name.rsplit('.', 1)
        mod = importlib.import_module(mod_name)
        func = getattr(mod, func_name)
        return func
    except Exception as e:
        raise Exception("Unknown executor name '{}'.\n{}".format(name, e))
```

basically, the diff is just `try` block.
currently, I'm monkey patching this and hope that this change is (a) lease invasive enough and (b) others might find it useful enough.
Please let me know if approved, I will be more than happy to make PR.

**Comment by Jesjf** [comment]
I have the same problem as Naoko. I want to handle all actions in a generic way using only one Python class. 
Has the contribution she offered been accepted? 
Has this issue been solved somehow? 
Is there another way to achieve my goal?

**Comment by kalimuthu123** [comment]
i want to interpret it with the live  database connection
